app/src/main/java/com/android/streetworkapp/ui/profile/AddFriend.kt

```diff
@@ -0,0 +1,114 @@
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.outlined.ArrowBack
+import androidx.compose.material3.Button
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.android.sample.R
+import com.android.streetworkapp.model.user.UserViewModel
+import com.android.streetworkapp.ui.navigation.NavigationActions
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun AddFriendScreen(
+    navigationActions: NavigationActions,
+    userViewModel: UserViewModel = viewModel(factory = UserViewModel.Factory)
+) {
+  // variable for outlined text
+  var id by remember { mutableStateOf("") }
+  // context for Toast
+  val context = LocalContext.current
+  // fake user ID (placeholder)
+  val uid = "user123"
+
+  Scaffold(
+      modifier = Modifier.testTag("addFriendScreen"),
+      topBar = {
+        // button to go back
+        TopAppBar(
+            title = { Text("Add a new friend", Modifier.testTag("addFriendTitle")) },
+            navigationIcon = {
+              IconButton(
+                  onClick = { navigationActions.goBack() }, Modifier.testTag("goBackButton")) {
+                    Icon(
+                        imageVector = Icons.AutoMirrored.Outlined.ArrowBack,
+                        contentDescription = "Back")
+                  }
+            })
+      },
+      content = { padding ->
+        Column(
+            modifier = Modifier.fillMaxSize().padding(padding).testTag("AddFriendColumn"),
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.spacedBy(20.dp, Alignment.CenterVertically)) {
+              Image(
+                  painter = painterResource(id = R.drawable.place_holder),
+                  contentDescription = "profile picture",
+                  modifier = Modifier.size(200.dp))
+
+              // button to activate NFC (don't work)
+              Button(onClick = {}, modifier = Modifier.size(220.dp, 50.dp).testTag("NFCButton")) {
+                Text(text = "Activate NFC", fontSize = 17.sp)
+              }
+
```

Comment:
You could separate this code from the UI screen by putting it in a @Composable function at the bottom of the file. This is important and was given as advice from the coaches in my branch :

> Small note for everyone, when writing UI code in jetpack compose, try limiting your composable functions in length. Create smaller, maybe more reusable composables and don't just put everything in a 'screen'.

Reply:
i think making a compodable for 3 line of code for a button is too extra

Reply:
Yes it seems extra, but when we will continue to implement this button to handle NFC, we will be happy that it is separated from the ui screen to prevent having a huge function (the main argument of the coach). 

This is the same case as with the `GoogleAuthButton`, and it would be important to do the same.